app/models/king.rb

```diff
@@ -13,6 +13,13 @@ def valid_move?(x, y)
     return false if delta_x(x_position, x) > 1
     return false if delta_y(y_position, y) > 1
 
+    # Prevent from moving into check
+    opposing_pieces = Piece.where('color != ? and captured = ?', color, false)
```

Comment:
You probably want `game.pieces.where` otherwise this is going to catch *any* piece from any game.